### DISCUSSION TITLE
印章文本检测DB++加svtr文字识别，碰到弯曲文本裁剪问题，这个到底要怎么解决，能联合识别吗

### DISCUSSION THREAD

**Discussion by shallow10** [PaddlePaddle/PaddleOCR#13621] [Q&A] [answered]
### Search before asking

- [X] I have searched the PaddleOCR [Docs](https://paddlepaddle.github.io/PaddleOCR/) and found no similar bug report.

- [X] I have searched the PaddleOCR [Issues](https://github.com/PaddlePaddle/PaddleOCR/issues) and found no similar bug report.

- [X] I have searched the PaddleOCR [Discussions](https://github.com/PaddlePaddle/PaddleOCR/discussions) and found no similar bug report.

### Bug

![Uploading image.png…]()

db++和文字识别都训练好了，就是用不了，弄了两周了看了各种办法都解决不了

### Environment

python3.8
paddle2.7.0
cuda11.8

### Minimal Reproducible Example

这个opencv裁剪到底怎么解决

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by shallow10** [comment]
[2024/08/08 14:52:31] ppocr DEBUG: 瑞安市南氏机械配件有限公司, 0.884
[2024/08/08 14:52:31] ppocr DEBUG: 法定代表人章, 1.000
[2024/08/08 14:52:31] ppocr DEBUG: 4021973548, 0.806
[2024/08/08 14:52:31] ppocr DEBUG: 法定代表人章, 1.000
[2024/08/08 14:52:31] ppocr DEBUG: 4021973548, 0.806
[2024/08/08 14:52:31] ppocr DEBUG: 寨荚丑, 0.999
Traceback (most recent call last):
  File "tools/infer/predict_system.py", line 272, in <module>
    main(args)
  File "tools/infer/predict_system.py", line 223, in main
    draw_img = draw_ocr_box_txt(
  File "E:\paddle\PaddleOCR-main\PaddleOCR-release-2.7\tools\infer\utility.py", line 453, in draw_ocr_box_txt
    img_right_text = draw_box_txt_fine((w, h), box, txt, font_path)
  File "E:\paddle\PaddleOCR-main\PaddleOCR-release-2.7\tools\infer\utility.py", line 490, in draw_box_txt_fine
    M = cv2.getPerspectiveTransform(pts1, pts2)
cv2.error: OpenCV(4.5.4) D:\a\opencv-python\opencv-python\opencv\modules\imgproc\src\imgwarp.cpp:3392: error: (-215:Assertion failed) src.checkVector(2, CV_32F) == 4 && dst.checkVector(2, CV_32F) == 4 in function 'cv::getPerspectiveTransform'

**Comment by shallow10** [comment]
有人能帮忙解决一下吗

**Comment by UserWangZz** [comment, marked as answer]
你是使用polygon格式还是quad格式的检测框？
polygon格式检测框可以尝试获取到检测框后进行一个矫正

**Reply by shallow10** [reply]
可以加个联系方式交流一下吗，我用的是poly的检测框，矫正里面svtr有个tps，但是一直这个报错，我不知道是透视变换那代码有问题还是tps没有生效，我弄了两周了各种办法都用了

**Reply by UserWangZz** [reply]
https://github.com/frotms/Curve-Text-Rectification-Using-Pairs-Of-Points
可以参考这个仓库的矫正，不过你需要分离出检测框的上下边。
下面是矫正结果参考
![image](https://github.com/user-attachments/assets/4040fa88-8f18-4f2b-be24-77580f4c424e)

**Reply by shallow10** [reply]
好的我试一下，我看到官方的效果是这样的，这个是怎么弄出来的？
![企业微信截图_17234289282744](https://github.com/user-attachments/assets/3d10192f-7010-4e08-aa33-792041296839)

**Comment by GreatV** [comment]
升级一下paddleocr版本

**Reply by shallow10** [reply]
大概要升级到什么版本？我现在是2.7.0

**Reply by shallow10** [reply]
试了更新到最新也不行
